game.js

```diff
@@ -31,11 +31,44 @@ class Game {
 
         this.log("Connecting to game.");
 
-        // TODO: Command line options to allow undo?
-        //
-        this.socket.on('game/' + game_id + '/undo_requested', (undodata) => {
-            this.log("Undo requested", JSON.stringify(undodata, null, 4));
-        });
+        if (config.UNDO && !config.UNDORANKED && !config.UNDOUNRANKED) {
+            this.socket.on('game/' + game_id + '/undo_requested_in_any_type_of_game', (undodata) => {
```

Comment:
I don't think you want to wrap this in `if (config.UNDO ...)` yet, the `if (...)` should probably be in the callback.